components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/UnsupportedAddonsAdapter.kt

```diff
@@ -40,18 +43,35 @@ class UnsupportedAddonsAdapter(
             }
 
         holder.removeButton.setOnClickListener {
+            holder.removeButton.visibility = View.GONE
             addonManager.uninstallAddon(addon,
                 onSuccess = {
-                    unsupportedAddonsAdapterDelegate.onUninstallSuccess()
+                    removeUninstalledAddonAtPosition(position)
                 },
                 onError = { addonId, throwable ->
                     unsupportedAddonsAdapterDelegate.onUninstallError(addonId, throwable)
+                    holder.removeButton.visibility = View.VISIBLE
                 })
         }
     }
 
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    @Suppress("TooGenericExceptionCaught")
+    internal fun removeUninstalledAddonAtPosition(position: Int) {
+        try {
+            val uninstalledAddon = _unsupportedAddons[position]
+            _unsupportedAddons.remove(uninstalledAddon)
+            notifyItemChanged(position)
+            unsupportedAddonsAdapterDelegate.onUninstallSuccess()
+        } catch (throwable: IndexOutOfBoundsException) {
```

Comment:
I'm a bit confuse of how we can get an `IndexOutOfBoundsException` as we are always deleting an item that is in bounds.

Reply:
Spamming the remove button would cause this. I set the visibility of the remove button to View.GONE to defend against this, but you just never know 🤷‍♂